homeassistant/components/solarlog/coordinator.py

```diff
@@ -84,8 +92,61 @@ async def _async_update_data(self) -> SolarlogData:
 
         _LOGGER.debug("Data successfully updated")
 
+        if self.solarlog.extended_data:
+            self._async_add_remove_devices(data)
+            _LOGGER.debug("Add_remove_devices finished")
+
         return data
 
+    def _async_add_remove_devices(self, data: SolarlogData) -> None:
+        """Add new devices, remove non-existing devices."""
+        if not self._devices_last_update:
+            self._devices_last_update = {
+                (k, self.solarlog.device_name(k)) for k in data.inverter_data
+            }
+            return
+
```

Comment:
```suggestion
```
I think you can remove this one, since it will just go through the whole process and still populate the list